File: peek_core_device/tuples/DeviceGpsLocationTuple.py
from vortex.Tuple import Tuple
from vortex.Tuple import TupleField
from vortex.Tuple import addTupleType

from peek_core_device._private.PluginNames import deviceTuplePrefix


@addTupleType
class DeviceGpsLocationTuple(Tuple):
    __tupleType__ = deviceTuplePrefix + "GpsLocationTuple"

    latitude: float = TupleField()
    longitude: float = TupleField()
    timestamp: int = TupleField(comment="timestamp in milliseconds")
    deviceId: str = TupleField()

File: peek_core_device/plugin-module/DeviceGpsLocationTuple.ts
import { addTupleType, Tuple } from "@synerty/vortexjs"
import { deviceTuplePrefix } from "./_private/PluginNames"


@addTupleType
export class DeviceGpsLocationTuple extends Tuple {
    public static readonly tupleName = deviceTuplePrefix + "GpsLocationTuple"
    
    // This field allows customer specific data, that peek doesn't need to work
    data: { [key: string]: any } = {}
    
    latitude: number
    longitude: number
    timestamp: number
    deviceToken: string
    
    constructor() {
        super(DeviceGpsLocationTuple.tupleName)
    }
}